Adding the Radklim reader with the stream YAML file

wael-mika opened the pull request:
## Description

<!-- Provide a brief summary of the changes introduced in this pull request. -->
This PR introduces a new data reader for the RADKLIM precipitation dataset, implemented using the base class interface. It also adds the corresponding YAML stream configuration under config/streams/streams_radklim/.

## Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ X ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

<!-- Link the Issue number this change addresses, ideally in one of the "magic format" such as Closes #XYZ -->
Closes #216 

<!-- Alternatively, explain the motivation behind the changes and the context in which they are being made. -->

## Code Compatibility

-   [ X ] I have performed a self-review of my code

### Code Performance and Testing

-   [ ] I ran the `uv run train` and (if necessary) `uv run evaluate` on a least one GPU node and it works 
-   [ ] If the new feature introduces modifications at the config level, I have made sure to have notified the other software developers through Mattermost and updated the paths in the `$WEATHER_GENERATOR_PRIVATE` directory

<!-- In case this affects the model sharding or other specific components please describe these here. -->

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ X ] I have not introduced new dependencies in the inference portion of the pipeline

<!-- List any new dependencies that are required for this change and the justification to add them. -->

### Documentation

-   [ X ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ X ] I have added comments to my code, particularly in hard-to-understand areas

<!-- Describe any major updates to the documentation -->

## Additional Notes

<!-- Include any additional information, caveats, or considerations that the reviewer should be aware of. -->
The stream configuration assumes the existence of RADKLIM-compatible reference data and normalization files.

The reader has been structured for easy extension to similar NetCDF-based datasets.

Further testing with training on multiple streams configurations and longer time windows is NECESSARY.

wael-mika committed:
Adding the Radklim reader with the stream YAML file

wael-mika committed:
Adding the Radklim reader with the stream YAML file

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,271 @@
+    """
+
+    # Channels used for source and target data
+    source_channels: list[str] = ["RR"]
```
This should be specified in the stream config and not hard coded, see `select_channels()` in data_reader_anemoi.

wael-mika replied:
Sure, I will adjust it, but as I mentioned, I need some guidance at the beginning.

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,271 @@
+    geoinfo_channels: list[str] = []
+
+    # Channel indices
+    source_idx: list[int] = [0]
```
This should be derived from the source_channels and not hard coded.

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,271 @@
+        nt, ny, nx, nvars = arr4.shape
+
+        # Validate channel indices
+        if not channels_idx:
```
This should be tested at the beginning of the function

wael-mika replied:
it makes sense, I will move it to the correct location.

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,271 @@
+        times = np.repeat(time_vals, self.points_per_slice)
+
+        # Apply nan filtering
+        valid = ~np.any(np.isnan(flat_vars[:, channels_idx]), axis=1)
```
No NaNs should be filtered here. This happens later.

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,271 @@
+        ds_win = self.ds.isel(time=slice(start, stop))
+
+        # Stack into (time, y, x, var) format
+        arr4 = (
```
Could you explain what arr4 is and how it comes out of ds_win

wael-mika replied:
This is the dataset for the selected time window
ds_win summary:
<xarray.Dataset> Size: 119MB
Dimensions:  (time: 6, y: 1100, x: 900)
Coordinates:
    lat      (time, y, x) float64 48MB ...
    lon      (time, y, x) float64 48MB ...
  * time     (time) datetime64[ns] 48B 2020-01-03T12:50:00 ... 2020-01-03T17:...
  * x        (x) float64 7kB -4.43e+05 -4.42e+05 -4.41e+05 ... 4.55e+05 4.56e+05
  * y        (y) float64 9kB -4.758e+06 -4.757e+06 ... -3.66e+06 -3.659e+06
Data variables:
    RR       (time, y, x) float32 24MB ...

This is the shape of the array when requesting the variable ('var', 'time', 'y', 'x')
arr4 (before transpose):
  dims : ('var', 'time', 'y', 'x')
  shape: (1, 6, 1100, 900)

Transpose to get (time, y, x, var) this was the requested shape from the older version, I can adapt to any shape you might need
arr4 (after transpose and .values):
  type : <class 'numpy.ndarray'>
  shape: (6, 1100, 900, 1)

clessig reviewed (commented):
Will try it out but here already some observations from looking at the code.

wael-mika committed:
updating the datareader and yaml file according to the comments

wael-mika committed:
updating the datareader and yaml file according to the comments

wael-mika committed:
updating the datareader and yaml file according to the comments

wael-mika committed:
updating the YAML file

wael-mika committed:
updating the YAML file

wael-mika committed:
Run ruff for the script

wael-mika committed:
Run ruff for the script

SavvasMel committed:
Introduce tuples instead for strings to avoid TypeError (#392)

clessig committed:
Exclude channels from src / target (#363)

* Exclude channels from src / target

* Simplified code and added comment that pattern matching is used

* Adding new stream config

* Fixing bug that led to error when accessing self.ds when dataset is empty

* Wokign on exlcude_source

* work in progress

* Fixing incorrect formating for logger (#388)

* Ruffed

* Refactored and cleaned up channel selection. Also added check that channels are not empty

* Cleaned channel parsing and selection

* Adjustments

* Removing asserts incompatible with empty dataset

---------

Co-authored-by: Christian Lessig <christian.lessig@ecwmf.int>

clessig commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,256 @@
+        with contextlib.suppress(Exception):
+            zarr.consolidate_metadata(mapper)
+
+        ds_full = xr.open_dataset(mapper, engine="zarr", consolidated=True)
```
That's the kerchunk version; this are the two magic lines:

```
 fs = fsspec.filesystem("reference", fo=kerchunk_ref)
mapper = fs.get_mapper("")
```

wael-mika replied:
Yes, it is this simple after creating the reference file and then using the little tricks of fsspec.

wael-mika commented on src/weathergen/datasets/data_reader_radklim.py:
```diff
@@ -0,0 +1,256 @@
+        return rdata
+
+
+def _clip_lat(lats: NDArray[np.floating]) -> NDArray[np.float32]:
```
Sure, we can do that as well

clessig replied:
Careful here: this function will not correctly convert from an arbitrary convention for spherical coords to the one we expect internally. For this reason, I would keep it local.

Wael, did you check that it is required at all?

TillHae committed:
add embed_dropout_rate to config v1 (#358)

tjhunter committed:
[402] adds checks to the pull request (#403)

* chanegs

* mistake

* mistake

* mistake

* changes

* doc

tjhunter committed:
Changes

Rebasing before trying the radklim data reader

shmh40 committed:
Introduce masking class and incorporate in TokenizerMasking (#383)

* creating masking class and adapting tokenizer_masking to use this class

* minor changes to masking.py and tokenizer_masking

* removed old tokenizer_masking

* include masking_strategy in default_config

* change ValueError to assert

* linting formatting changes files

* further linting of docstrings

* create mask_source and mask_target in Masker, and update tokenizer_masking to use these, then style improvements

* linted masking, tokenizer_masking

* modify masker, rng and perm_sel now part of class, remove extra masking_rate, update comments, remove archived class

* remove check if all masked, not masked

* remove self.masking_rate from MultiStreamDS class, and masking args from batchify_source

* update tokenizer utils with description of idx_ord_lens in comment

* remove masking args from batchify_, perm_sel removed now internal to Masker class, remove handling special cases of masking (all masked)

* adding masking_strategy: to config

* remove unused mentions of masking_combination

* removed comment about streams

* changed assert to check self perm_sel is not None

* ruff masking, tokenizer_masking

* Ruffed

* Added warning to capture corner case, likely due to incorrect user settings.

* Fixed incorrect call twice

* Fixed missing conditional for logger statement

* Required changes for better handling of rngs

* Improved handling of rngs

* Improved handling of rng

---------

Co-authored-by: Christian Lessig <christian.lessig@ecmwf.int>

wael-mika committed:
Update for TIM: radklim data reader, config and data-sampler logic

wael-mika committed:
Update for TIM: radklim data reader, config and data-sampler logic

wael-mika committed:
Update for TIM: radklim data reader, config and data-sampler logic

kacpnowak committed:
Implement per-channel logging (#283)

* Fix bug with seed being divided by 0 for worker ID=0

* Fix bug causing crash when secrets aren't in private config

* Implement logging losses per channel

* Fix issue with empty targets

* Rework loss logging

* ruff

* Remove computing max_channels

* Change variables names

* ruffed

* Remove redundant enumerations

* Use stages for logging

* Add type hints

* Apply the review

* ruff

* fix

* Fix type hints

* ruff

---------

Co-authored-by: Tim Hunter <tim.hunter@ecmwf.int>

tjhunter committed:
[346] Passing options through the slurm script (#400)

* changes

* fixes

tjhunter committed:
changes

Continue rebase

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes (#422)

tjhunter committed:
[420] assign (#424)

* changes

* changes

clessig committed:
Fix shuffle for validation (#413)

shuffle=False for validation at the moment. Should be True to have an unbiased MC estimator over the full val set.

tjhunter committed:
[420] Tjh/dev/420 gh assign (#425)

* changes

* changes

* change

sophie-xhonneux committed:
Clessig/develop/fix instability (#419)

* - Avoid time encoding is 0
- eps in layer norms to 10^-3
- bf16

* Fixed incorrect cast

* Make the attention dtype and norm eps configurable

* Fix gitignore and add config files

* Shuffle config files into sensible folders

* Try fp16

* Fix some missing hardcoded

* recover num_ranks from previous run to calculate epoch_base (#317)

* recover num_ranks from previous run to calculate epoch_base

* set email settings for commits

* addressing Tim's comment

* make ruff happy

* improve style

* changes (#385)

Linter rule so np.ndarray is not used as type

* changed the script name from evaluate to inference as it simply gener… (#376)

* changed the script name from evaluate to inference as it simply generate infer samples

* changed evaluate to inference in the main scripts and corresponding calls in the config

* update the main function for the inference script

* changed evaluate to inference also in docstring, unit test scripts, and integration test scripts

---------

Co-authored-by: Patnala,Ankit <a.patnala@fz-juelich.de>

* Introduce tuples instead for strings to avoid TypeError (#392)

* Exclude channels from src / target (#363)

* Exclude channels from src / target

* Simplified code and added comment that pattern matching is used

* Adding new stream config

* Fixing bug that led to error when accessing self.ds when dataset is empty

* Wokign on exlcude_source

* work in progress

* Fixing incorrect formating for logger (#388)

* Ruffed

* Refactored and cleaned up channel selection. Also added check that channels are not empty

* Cleaned channel parsing and selection

* Adjustments

* Removing asserts incompatible with empty dataset

---------

Co-authored-by: Christian Lessig <christian.lessig@ecwmf.int>

* add embed_dropout_rate to config v1 (#358)

* [402] adds checks to the pull request (#403)

* chanegs

* mistake

* mistake

* mistake

* changes

* doc

* Introduce masking class and incorporate in TokenizerMasking (#383)

* creating masking class and adapting tokenizer_masking to use this class

* minor changes to masking.py and tokenizer_masking

* removed old tokenizer_masking

* include masking_strategy in default_config

* change ValueError to assert

* linting formatting changes files

* further linting of docstrings

* create mask_source and mask_target in Masker, and update tokenizer_masking to use these, then style improvements

* linted masking, tokenizer_masking

* modify masker, rng and perm_sel now part of class, remove extra masking_rate, update comments, remove archived class

* remove check if all masked, not masked

* remove self.masking_rate from MultiStreamDS class, and masking args from batchify_source

* update tokenizer utils with description of idx_ord_lens in comment

* remove masking args from batchify_, perm_sel removed now internal to Masker class, remove handling special cases of masking (all masked)

* adding masking_strategy: to config

* remove unused mentions of masking_combination

* removed comment about streams

* changed assert to check self perm_sel is not None

* ruff masking, tokenizer_masking

* Ruffed

* Added warning to capture corner case, likely due to incorrect user settings.

* Fixed incorrect call twice

* Fixed missing conditional for logger statement

* Required changes for better handling of rngs

* Improved handling of rngs

* Improved handling of rng

---------

Co-authored-by: Christian Lessig <christian.lessig@ecmwf.int>

* Make the attention dtype and norm eps configurable

* Final default config

* Implement per-channel logging (#283)

* Fix bug with seed being divided by 0 for worker ID=0

* Fix bug causing crash when secrets aren't in private config

* Implement logging losses per channel

* Fix issue with empty targets

* Rework loss logging

* ruff

* Remove computing max_channels

* Change variables names

* ruffed

* Remove redundant enumerations

* Use stages for logging

* Add type hints

* Apply the review

* ruff

* fix

* Fix type hints

* ruff

---------

Co-authored-by: Tim Hunter <tim.hunter@ecmwf.int>

* [346] Passing options through the slurm script (#400)

* changes

* fixes

* - Avoid time encoding is 0
- eps in layer norms to 10^-3
- bf16

* Make the attention dtype and norm eps configurable

* Fix gitignore and add config files

* Clean up configs for PR

* Clean up the forgotten HEAD

* Apply ruff formatting

* Organize imports

* Add mlp norm eps to embed targets and pred adapter

* Add comment

---------

Co-authored-by: Christian Lessig <christian.lessig@ecmwf.int>
Co-authored-by: Julian Kuehnert <Jubeku@users.noreply.github.com>
Co-authored-by: Timothy Hunter <tim.hunter@ecmwf.int>
Co-authored-by: ankitpatnala <ankitpatnala@gmail.com>
Co-authored-by: Patnala,Ankit <a.patnala@fz-juelich.de>
Co-authored-by: Savvas Melidonis <79579567+SavvasMel@users.noreply.github.com>
Co-authored-by: Christian Lessig <christian.lessig@ecwmf.int>
Co-authored-by: Till Hauer <till@web-hauer.de>
Co-authored-by: Seb Hickman <56727418+shmh40@users.noreply.github.com>
Co-authored-by: Kacper Nowak <kacper.nowak@awi.de>

tjhunter committed:
Revert "Implement per-channel logging (#283)" (#434)

This reverts commit 989ab6e1d6e8c0f69594414c7733adf30acd1c54.

wael-mika committed:
updating the data reader to use lazy loading to avoid the forking error

wael-mika committed:
updating the data reader to use lazy loading to avoid the forking error

wael-mika committed:
updating the data reader to use lazy loading to avoid the forking error

Rebase before merging

wael-mika committed:
ruff formated

wael-mika committed:
ruff formated

rebase

wael-mika committed:
ruff formated

tjhunter committed:
changes

wael-mika committed:
multi_stream_data_sampler.py fixed by ruff

wael-mika committed:
multi_stream_data_sampler.py fixed by ruff

wael-mika committed:
multi_stream_data_sampler.py fixed by ruff

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

Commiting changes before rebase

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

tjhunter committed:
changes

kacpnowak committed:
Fix FESOM datareader and int overflow  (#417)

* Fix indexing in DataReaderFesom

* Enforce using only int64 in data loading

* ruff

* ruff2

* Review

* Change int64 back to int32

tjhunter committed:
changes (#462)

clessig committed:
Fix incorrect handling of empty window (which triggered problem in IO writing code). (#447)

clessig committed:
Update default_config.yml (#446)

analysis_streams_output is missing, which leads to error with val_initial=True and log_validation > 0.

mlangguth89 committed:
Re-enabled option to run plot_training as script and fixed -rf argument (#444)

* Re-enabled option to runplot_training as script and removed relative path as default from mutually-exclusive argument -rf.

* Ruffed code.

* Ruff check fix.

* Rename flags for parsing configuration and fixed default handling for standard config YAML-file.

clessig committed:
fix era5 config (#473)

Adding z back in

tjhunter committed:
[251] Merge new IO class (#469)

* Implement mock IO (#336)

* Adapt score class score class (#339)

* Implement mock IO

* Adapt score class

* Removing unused file (#349)

* remove database folder (#355)

* Small change - CI - pinning the version of formatting (#361)

* changes

* changes

* Update INSTALL.md

* Update INSTALL.md

* Fixed Exxx lint issues (#284)

* Rebased to the latest changes and linted new changes

* addressed review comments

* addressed review comments

* Linted the latest changes.

* corrected the formating

* corrected the formating

* configured ruff to use LF line endings in pyproject.toml

* [357] Sub-package for evaluation (#359)

* working

* changes

* removing deps from non-core project

* changes

* fixes

* comments

* Iluise quick fix stac (#374)

* remove database folder

* fix database

* Simplifying workflow for plot_training (#368)

* Simplifying workflow for plot_training

* Ruffed

* Working on implementing exclude_source

* Remove unused code

* Fixed ruff issue

* Fixing bug in lat handling (377) (#378)

* Fixing bug in lat handling

* Added comment

---------

Co-authored-by: Seb Hickman <56727418+shmh40@users.noreply.github.com>

* recover num_ranks from previous run to calculate epoch_base (#317)

* recover num_ranks from previous run to calculate epoch_base

* set email settings for commits

* addressing Tim's comment

* make ruff happy

* improve style

* changes (#385)

Linter rule so np.ndarray is not used as type

* changed the script name from evaluate to inference as it simply gener… (#376)

* changed the script name from evaluate to inference as it simply generate infer samples

* changed evaluate to inference in the main scripts and corresponding calls in the config

* update the main function for the inference script

* changed evaluate to inference also in docstring, unit test scripts, and integration test scripts

---------

Co-authored-by: Patnala,Ankit <a.patnala@fz-juelich.de>

* Introduce tuples instead for strings to avoid TypeError (#392)

* Exclude channels from src / target (#363)

* Exclude channels from src / target

* Simplified code and added comment that pattern matching is used

* Adding new stream config

* Fixing bug that led to error when accessing self.ds when dataset is empty

* Wokign on exlcude_source

* work in progress

* Fixing incorrect formating for logger (#388)

* Ruffed

* Refactored and cleaned up channel selection. Also added check that channels are not empty

* Cleaned channel parsing and selection

* Adjustments

* Removing asserts incompatible with empty dataset

---------

Co-authored-by: Christian Lessig <christian.lessig@ecwmf.int>

* add embed_dropout_rate to config v1 (#358)

* [402] adds checks to the pull request (#403)

* chanegs

* mistake

* mistake

* mistake

* changes

* doc

* Introduce masking class and incorporate in TokenizerMasking (#383)

* creating masking class and adapting tokenizer_masking to use this class

* minor changes to masking.py and tokenizer_masking

* removed old tokenizer_masking

* include masking_strategy in default_config

* change ValueError to assert

* linting formatting changes files

* further linting of docstrings

* create mask_source and mask_target in Masker, and update tokenizer_masking to use these, then style improvements

* linted masking, tokenizer_masking

* modify masker, rng and perm_sel now part of class, remove extra masking_rate, update comments, remove archived class

* remove check if all masked, not masked

* remove self.masking_rate from MultiStreamDS class, and masking args from batchify_source

* update tokenizer utils with description of idx_ord_lens in comment

* remove masking args from batchify_, perm_sel removed now internal to Masker class, remove handling special cases of masking (all masked)

* adding masking_strategy: to config

* remove unused mentions of masking_combination

* removed comment about streams

* changed assert to check self perm_sel is not None

* ruff masking, tokenizer_masking

* Ruffed

* Added warning to capture corner case, likely due to incorrect user settings.

* Fixed incorrect call twice

* Fixed missing conditional for logger statement

* Required changes for better handling of rngs

* Improved handling of rngs

* Improved handling of rng

---------

Co-authored-by: Christian Lessig <christian.lessig@ecmwf.int>

* Implement per-channel logging (#283)

* Fix bug with seed being divided by 0 for worker ID=0

* Fix bug causing crash when secrets aren't in private config

* Implement logging losses per channel

* Fix issue with empty targets

* Rework loss logging

* ruff

* Remove computing max_channels

* Change variables names

* ruffed

* Remove redundant enumerations

* Use stages for logging

* Add type hints

* Apply the review

* ruff

* fix

* Fix type hints

* ruff

---------

Co-authored-by: Tim Hunter <tim.hunter@ecmwf.int>

* [346] Passing options through the slurm script (#400)

* changes

* fixes

* refactor `validation_io.write_validation` to make it more readable

* remove legacy code `validation_io.read_validation`

* encapsulate artifact path logic in config module

* remove redundant attribute `Trainer.path_run`

* use config to look up base_path in `write_validation`

* remove unused `write_validation` args: `base_path`, `rank`

* ensure correct type for pathes

* remove streams initialization from `Trainer`

* remove path logic from `Trainer.save_model`

* simplify conditional

* rename mock io module

* update uv to include dask

* Implement io module to support reading/writing model output

* implement new validation_io routine

* use new write_validation routine

* remove unused code

* rename output routine to `write_output`

* ruffed and added comments

* fixed annotation

* use simple __init__ method for `OutputItem` instead of dataclasses magic

* address reviewers comments

* rename method

* add simple docstrings

* ruffed

* typehint fixes

* refactor names

* update comments and typehints, dont import pytorch

* remove `__post_init__` methods, cache properties

* fixes and integration test

* final fixes :)

* changes

* changes

* changes

* changes

* changes

* more work

* changes

* changes

* changes

* ruffed

* ruffed

* improve logging and comments

* Update to score-class according to internal discussions and feedback in PR.

* Add license header.

* Ruffed code.

* Update to score-class according to internal discussions and feedback in PR.

* Add license header.

* Ruffed code.

* Add doc-string to call-method and provide example usage for efficient graph-construction.

* Some fixes to score-class.

* Some fixes to handling aggregation dimension.

* Add missing import of MockIO.

* changes

* changes

* removing the scores

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

* changes

---------

Co-authored-by: Kacper Nowak <kacper.nowak@awi.de>
Co-authored-by: Christian Lessig <christian.lessig@ecmwf.int>
Co-authored-by: iluise <72020169+iluise@users.noreply.github.com>
Co-authored-by: Sindhu-Vasireddy <98752594+Sindhu-Vasireddy@users.noreply.github.com>
Co-authored-by: Seb Hickman <56727418+shmh40@users.noreply.github.com>
Co-authored-by: Julian Kuehnert <Jubeku@users.noreply.github.com>
Co-authored-by: ankitpatnala <ankitpatnala@gmail.com>
Co-authored-by: Patnala,Ankit <a.patnala@fz-juelich.de>
Co-authored-by: Savvas Melidonis <79579567+SavvasMel@users.noreply.github.com>
Co-authored-by: Christian Lessig <christian.lessig@ecwmf.int>
Co-authored-by: Till Hauer <till@web-hauer.de>
Co-authored-by: Simon Grasse <s.grasse@fz-juelich.de>
Co-authored-by: Michael <m.langguth@fz-juelich.de>

tjhunter committed:
[459] Attempt to fix ruff differences (#463)

* changes

* debug

* changes

* changes

clessig committed:
Update pyproject.toml (#457)

Jubeku committed:
Continue training through slurm script (#395)

* train_continue via slurm

* using __main__ as entry point for slurm script

* reverting config files to match base branch

* reverting config files to match base branch

* removing param_sum control logging before and after loading of model weights

* run ruff

* check whether from_run_id is in arguments

* trigger PR check

* remove block to set reuse_run_id=True

---------

Co-authored-by: Julian Kuehnert <julian.kuehnert@ecwmf.int>

kctezcan committed:
added the .python_version file set to python 3.12 (#482)

Co-authored-by: Kerem Can Tezcan <ktezcan0@login07.leonardo.local>

tjhunter committed:
script (#489)

enssow committed:
Remove print statements for logging (#421) (#439)

* first change

* removed all prints

* changed model.py back

* adding comments and fixes@

* added ruff fixes

* reverting files for PR

* ruff fixes

* removing run_id.py

* formatting changes

* changing comments in check_gh_issue script

---------

Co-authored-by: owens1 <owens1@jwlogin09.juwels>
Co-authored-by: Timothy Hunter <tim.hunter@ecmwf.int>

sophie-xhonneux committed:
Rename batchsize to batchsize_per_gpu (#475)

* Rename batchsize to batchsize_per_gpu

* Fix ruff stuff

tjhunter committed:
fix (#490)

shmh40 committed:
add polar orbiters and abi-goes to the stac database (#426)

* testing adding metopa and metopb as placeholder drafts to stac database

* added the actual json files because I think we have to

* updated metopa metopb jsons and ets

* add fy3 and update metops

* updated names of metops

* updated metopb untarred size inodes and end date

* update names to instrument, satellite

* add untarred data size and inodes for metopa

* updated to oscar naming, with format platform, instrument, and added fengyun satellites

* update size and inodes of fy3c mwhs

* add fengyun jsons, missing before, and update unique ids of metopa and b

* add processing_level field to metopa as a test

* adding processing level field

* fix up processing level

* updated jsons and jsonnets for provenance

* actually include provenance

* updated to include processor and provider, remove provenance

* add abi-goes

* fix abi goes geometry

* fix latitude and longitude

* fix typo

* hopefully this time lat is right..

* update catalogue json for develop

* check catalogue on this branch

* jsonneted for develop

---------

Co-authored-by: iluise <luise.ilaria@gmail.com>

MatKbauer committed:
Added naming convention checks to lint (#501)

* Added naming convention checks to lint

* Implemented python naming conventions and corrected code accordingly

---------

Co-authored-by: Matthias Karlbauer <ecm1575@ac6-102.bullx>

wael-mika committed:
Checkpoint for the datareader with the memory and masking issues

wael-mika committed:
Merge branch 'wm/dev/radklim' of https://github.com/wael-mika/WeatherGenerator into wm/dev/radklim

MatKbauer committed:
Correct the in-code-names for rotation matrices (#516)

* Added naming convention checks to lint

* Implemented python naming conventions and corrected code accordingly

* Corrected renaming of rotation matrices from R to rot instead of to r

---------

Co-authored-by: Matthias Karlbauer <ecm1575@ac6-102.bullx>

jpolz committed:
extend format string and timedelta to days (#499)

* extend format string and timedelta to days

* replace with pd.to_timedelta

* import pandas

* ruff

* enforce "HH:MM:SS" format

* ruff

mlangguth89 committed:
Mlangguth/develop/issue 251 (#495)

* Add score-class to evaluate-package.

* Add score-class to evaluate-package.

* Lintered and ruffed code.

* Add fix to io.py and update dependencies in common.

* Several small fixes to score-class and fast evaluation.

* Add utils for evaluate.

* Moved to_list to utils and improved doc-strings.

* Improve several doc-strings, avoid formatting of logger and other changes from PR review.

* Add xhistogram and xskillscore to dependencies of evaluate.

* Ruffed code.

* Lintered code.

* Fix incorrect retrieval of validation batch size in validation IO.

* Final minor changes to argument-names

tjhunter committed:
changes (#471)

Sindhu-Vasireddy committed:
Updated to camel case. (#445)

* Updated to camel case.

* Fixed formatting.

tjhunter committed:
Revert "Updated to camel case. (#445)" (#530)

This reverts commit 4a8bd49067d86c8c9dd2930544d52cb9db8577af.

tjhunter committed:
[327] Script to create the links to output directories (results, ...) (#528)

* changes

* fixes

* slash

* slash

* checks

* checks

wael-mika committed:
branch major update before debuging with ecmwf

Continue rebase before merging

wael-mika committed:
Apply ruff/Black formatting fixes

wael-mika committed:
some tweaks before debugging

wael-mika committed:
Update config_radklim.yml

Use relative path in the config file
